src/lib/sedlexing.ml

```diff
@@ -228,7 +268,28 @@ module Latin1 = struct
       finished = true;
     }
 
-  let from_channel ic = from_gen (gen_of_channel ic)
+  let from_channel ic =
```

Comment:
All the `from_channel` functions are pretty similar, can we factor out the common logic?

Such as
```
let make_from_channel ~buffer_multiplier ~min_bytes_per_uchar ~read_uchar_from_channel =
  fun ic ->
    let t = Chan.create ic (chunk_size * buffer_multiplier) in
    let bo = ref opt_bo in
    let refill buf pos len =
      let rec loop i =
        if i = len then i
        else (
          match
            (* we refill our bytes buffer only if we haven't refilled any uchar yet. *)
            let can_refill = i = 0 in
            Chan.ensure t ~can_refill min_bytes_per_uchar;
						read_uchar_from_channel t
          with
            | c ->
                buf.(pos + i) <- c;
                loop (i + 1)
            | exception Chan.Missing_input ->
                if i = 0 && Chan.available t > 0 then raise MalFormed;
                i)
      in
      loop 0
    in
    create refill
```

Reply:
I've made the change